### PR TITLE
fix: select field conflict with WordPress style

### LINKED ISSUE #133: fix: select field conflict with WordPress style
# Bug Report

## Developer Story
As a developer, I want the select field to look like as in Figma design.

From Figma
image

image

From Builder
image

## Acceptance Criteria
- [ ] Select field style looks good.

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This Pr will resolve #133 

## How Has This Been Tested?
before
image
image

After
image
image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@knowler Few noticeable changes
1. I update logic in the `Label` component which allows us to render it only if `children` exist because in a few cases we only need the input field without a label.
ref: https://github.com/impress-org/givewp-campaigns/pull/134/commits/c57aba4dc8cbb230ecd019d5de84fd8718ac2be5

2. We already had prefix support in the Select component. In this pr, I also added support for suffix and label.

image

**Comment by knowler** [comment]
Re: 1

That's what the `labelHidden` props on Label is for. Every form control should have a label.

File: resources/css/components/inputs.css [hunk 1]
```diff
@@ -7,3 +7,9 @@ input[type='radio'] {
 
 	@apply border-gray-500;
 }
+
+select,
+input {
+	box-shadow: none;
```
Comment (knowler) [inline resources/css/components/inputs.css]: Current `input[type='radio']` uses the `box-shadow` it’s inheriting from the WordPress admin styles. Perhaps consider: `input:not([type='radio'])`. Also, you might want to check if the checkboxes are doing the same.

Ideally, our components really shouldn’t be relying on the WordPress admin styles (or anything from any other context except the browser reset) and should clarify what they need themselves, but we’re living the prototype-life. 😝

File: resources/js/components/select/index.jsx [hunk 2]
```diff
@@ -3,53 +3,102 @@
...
-					<FontAwesomeIcon icon={ faCaretUp } className="absolute top-0 right-0 mb-2 text-gray-700" />
-					<FontAwesomeIcon icon={ faCaretDown } className="absolute top-0 right-0 mt-2 text-gray-700" />
+		<div>
+			<Label
+				htmlFor={ id || name }
+				hidden={ labelHidden }
+				instructions={ instructions } >
+				{ label }
+			</Label>
+			<div className="w-full bg-white border border-gray-400 hover:border-gray-500 px-3 py-2">
```
Comment (knowler) [inline resources/js/components/select/index.jsx]: Screenshot 2019-10-18 09 53 10

We will want a rounded border as well.

File: resources/css/components/inputs.css (hunk 1 above)
Comment (ravinderk) [reply resources/css/components/inputs.css]: @knowler  Good suggestion but `input` has less preference than `input[type='radio']`, so the default WordPress style of the radio input field and the checkbox input field will not be affected by this CSS.

**Pull request merged**